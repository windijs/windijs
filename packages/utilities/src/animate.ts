import { animateHandler, createUtility } from "@windijs/core";
import { backInDownKeyframes, backInLeftKeyframes, backInRightKeyframes, backInUpKeyframes, backOutDownKeyframes, backOutLeftKeyframes, backOutRightKeyframes, backOutUpKeyframes, bounceInDownKeyframes, bounceInKeyframes, bounceInLeftKeyframes, bounceInRightKeyframes, bounceInUpKeyframes, bounceKeyframes, bounceOutDownKeyframes, bounceOutKeyframes, bounceOutLeftKeyframes, bounceOutRightKeyframes, bounceOutUpKeyframes, bubbleKeyframes, fadeInBottomLeftKeyframes, fadeInBottomRightKeyframes, fadeInDownBigKeyframes, fadeInDownKeyframes, fadeInKeyframes, fadeInLeftBigKeyframes, fadeInLeftKeyframes, fadeInRightBigKeyframes, fadeInRightKeyframes, fadeInTopLeftKeyframes, fadeInTopRightKeyframes, fadeInUpBigKeyframes, fadeInUpKeyframes, fadeOutDownBigKeyframes, fadeOutDownKeyframes, fadeOutKeyframes, fadeOutLeftBigKeyframes, fadeOutLeftKeyframes, fadeOutRightBigKeyframes, fadeOutRightKeyframes, fadeOutUpBigKeyframes, fadeOutUpKeyframes, flashKeyframes, flipInXKeyframes, flipInYKeyframes, flipKeyframes, flipOutXKeyframes, flipOutYKeyframes, headShakeKeyframes, heartBeatKeyframes, hingeKeyframes, jackInTheBoxKeyframes, jelloKeyframes, lightSpeedInLeftKeyframes, lightSpeedInRightKeyframes, lightSpeedOutLeftKeyframes, lightSpeedOutRightKeyframes, pingKeyframes, pulseKeyframes, rollInKeyframes, rollOutKeyframes, rotateInDownLeftKeyframes, rotateInDownRightKeyframes, rotateInKeyframes, rotateInUpLeftKeyframes, rotateInUpRightKeyframes, rotateOutDownLeftKeyframes, rotateOutDownRightKeyframes, rotateOutKeyframes, rotateOutUpLeftKeyframes, rotateOutUpRightKeyframes, rubberBandKeyframes, shakeXKeyframes, shakeYKeyframes, shockKeyframes, slideInDownKeyframes, slideInLeftKeyframes, slideInRightKeyframes, slideInUpKeyframes, slideOutDownKeyframes, slideOutLeftKeyframes, slideOutRightKeyframes, slideOutUpKeyframes, spinKeyframes, swingKeyframes, tadaKeyframes, wobbleKeyframes, zoomInDownKeyframes, zoomInKeyframes, zoomInLeftKeyframes, zoomInRightKeyframes, zoomInUpKeyframes, zoomOutDownKeyframes, zoomOutKeyframes, zoomOutLeftKeyframes, zoomOutRightKeyframes, zoomOutUpKeyframes } from "@windijs/config";

export default createUtility("animate")
  .use(animateHandler("none", "none"))
  .use(animateHandler("spin", "spin 1s linear infinite", spinKeyframes))
  .use(animateHandler("ping", "ping 1s cubic-bezier(0, 0, 0.2, 1) infinite", pingKeyframes))
  .use(animateHandler("pulse", "pulse 2s cubic-bezier(0.4, 0, 0.6, 1) infinite", pulseKeyframes))
  .use(animateHandler("bounce", "bounce 1s infinite", bounceKeyframes))
  .use(animateHandler("shock", { animation: "shock", transformOrigin: "center bottom" }, shockKeyframes))
  .use(animateHandler("flash", "flash", flashKeyframes))
  .use(animateHandler("bubble", "bubble", bubbleKeyframes))
  .use(animateHandler("rubberBand", "rubberBand", rubberBandKeyframes))
  .use(animateHandler("shakeX", "shakeX", shakeXKeyframes))
  .use(animateHandler("shakeY", "shakeY", shakeYKeyframes))
  .use(animateHandler("headShake", "headShake 1s ease-in-out", headShakeKeyframes))
  .use(animateHandler("swing", { animation: "swing", transformOrigin: "top center" }, swingKeyframes))
  .use(animateHandler("tada", "tada", tadaKeyframes))
  .use(animateHandler("wobble", "wobble", wobbleKeyframes))
  .use(animateHandler("jello", "jello", jelloKeyframes))
  .use(animateHandler("heartBeat", "heartBeat 1s ease-in-out", heartBeatKeyframes))
  .use(animateHandler("hinge", "hinge 2s", hingeKeyframes))
  .use(animateHandler("jackIn", "jackInTheBox", jackInTheBoxKeyframes))
  .use(animateHandler("lightSpeedInLeft", "lightSpeedInLeft", lightSpeedInLeftKeyframes))
  .use(animateHandler("lightSpeedInRight", "lightSpeedInRight", lightSpeedInRightKeyframes))
  .use(animateHandler("lightSpeedOutLeft", "lightSpeedOutLeft", lightSpeedOutLeftKeyframes))
  .use(animateHandler("lightSpeedOutRight", "lightSpeedOutRight", lightSpeedOutRightKeyframes))
  .use(animateHandler("flip", { animation: "flip", backfaceVisibility: "visible" }, flipKeyframes))
  .use(animateHandler("flipInX", { animation: "flipInX", backfaceVisibility: "visible" }, flipInXKeyframes))
  .use(animateHandler("flipInY", { animation: "flipInY", backfaceVisibility: "visible" }, flipInYKeyframes))
  .use(animateHandler("flipOutX", { animation: "flipOutX", backfaceVisibility: "visible" }, flipOutXKeyframes))
  .use(animateHandler("flipOutY", { animation: "flipOutY", backfaceVisibility: "visible" }, flipOutYKeyframes))
  .use(animateHandler("rotateIn", "rotateIn", rotateInKeyframes))
  .use(animateHandler("rotateInDownLeft", "rotateInDownLeft", rotateInDownLeftKeyframes))
  .use(animateHandler("rotateInDownRight", "rotateInDownRight", rotateInDownRightKeyframes))
  .use(animateHandler("rotateInUpLeft", "rotateInUpLeft", rotateInUpLeftKeyframes))
  .use(animateHandler("rotateInUpRight", "rotateInUpRight", rotateInUpRightKeyframes))
  .use(animateHandler("rotateOut", "rotateOut", rotateOutKeyframes))
  .use(animateHandler("rotateOutDownLeft", "rotateOutDownLeft", rotateOutDownLeftKeyframes))
  .use(animateHandler("rotateOutDownRight", "rotateOutDownRight", rotateOutDownRightKeyframes))
  .use(animateHandler("rotateOutUpLeft", "rotateOutUpLeft", rotateOutUpLeftKeyframes))
  .use(animateHandler("rotateOutUpRight", "rotateOutUpRight", rotateOutUpRightKeyframes))
  .use(animateHandler("rollIn", "rollIn", rollInKeyframes))
  .use(animateHandler("rollOut", "rollOut", rollOutKeyframes))
  .use(animateHandler("zoomIn", "zoomIn", zoomInKeyframes))
  .use(animateHandler("zoomInDown", "zoomInDown", zoomInDownKeyframes))
  .use(animateHandler("zoomInLeft", "zoomInLeft", zoomInLeftKeyframes))
  .use(animateHandler("zoomInRight", "zoomInRight", zoomInRightKeyframes))
  .use(animateHandler("zoomInUp", "zoomInUp", zoomInUpKeyframes))
  .use(animateHandler("bounceIn", "bounceIn 750ms", bounceInKeyframes))
  .use(animateHandler("bounceInDown", "bounceInDown", bounceInDownKeyframes))
  .use(animateHandler("bounceInLeft", "bounceInLeft", bounceInLeftKeyframes))
  .use(animateHandler("bounceInRight", "bounceInRight", bounceInRightKeyframes))
  .use(animateHandler("bounceInUp", "bounceInUp", bounceInUpKeyframes))
  .use(animateHandler("bounceOut", "bounceOut 750ms", bounceOutKeyframes))
  .use(animateHandler("bounceOutDown", "bounceOutDown", bounceOutDownKeyframes))
  .use(animateHandler("bounceOutLeft", "bounceOutLeft", bounceOutLeftKeyframes))
  .use(animateHandler("bounceOutRight", "bounceOutRight", bounceOutRightKeyframes))
  .use(animateHandler("bounceOutUp", "bounceOutUp", bounceOutUpKeyframes))
  .use(animateHandler("zoomOut", "zoomOut", zoomOutKeyframes))
  .use(animateHandler("zoomOutDown", "zoomOutDown", zoomOutDownKeyframes))
  .use(animateHandler("zoomOutLeft", "zoomOutLeft", zoomOutLeftKeyframes))
  .use(animateHandler("zoomOutRight", "zoomOutRight", zoomOutRightKeyframes))
  .use(animateHandler("zoomOutUp", "zoomOutUp", zoomOutUpKeyframes))
  .use(animateHandler("slideInDown", "slideInDown", slideInDownKeyframes))
  .use(animateHandler("slideInLeft", "slideInLeft", slideInLeftKeyframes))
  .use(animateHandler("slideInRight", "slideInRight", slideInRightKeyframes))
  .use(animateHandler("slideInUp", "slideInUp", slideInUpKeyframes))
  .use(animateHandler("slideOutDown", "slideOutDown", slideOutDownKeyframes))
  .use(animateHandler("slideOutLeft", "slideOutLeft", slideOutLeftKeyframes))
  .use(animateHandler("slideOutRight", "slideOutRight", slideOutRightKeyframes))
  .use(animateHandler("slideOutUp", "slideOutUp", slideOutUpKeyframes))
  .use(animateHandler("fadeIn", "fadeIn", fadeInKeyframes))
  .use(animateHandler("fadeInDown", "fadeInDown", fadeInDownKeyframes))
  .use(animateHandler("fadeInDownBig", "fadeInDownBig", fadeInDownBigKeyframes))
  .use(animateHandler("fadeInLeft", "fadeInLeft", fadeInLeftKeyframes))
  .use(animateHandler("fadeInLeftBig", "fadeInLeftBig", fadeInLeftBigKeyframes))
  .use(animateHandler("fadeInRight", "fadeInRight", fadeInRightKeyframes))
  .use(animateHandler("fadeInRightBig", "fadeInRightBig", fadeInRightBigKeyframes))
  .use(animateHandler("fadeInUp", "fadeInUp", fadeInUpKeyframes))
  .use(animateHandler("fadeInUpBig", "fadeInUpBig", fadeInUpBigKeyframes))
  .use(animateHandler("fadeInTopLeft", "fadeInTopLeft", fadeInTopLeftKeyframes))
  .use(animateHandler("fadeInTopRight", "fadeInTopRight", fadeInTopRightKeyframes))
  .use(animateHandler("fadeInBottomLeft", "fadeInBottomLeft", fadeInBottomLeftKeyframes))
  .use(animateHandler("fadeInBottomRight", "fadeInBottomRight", fadeInBottomRightKeyframes))
  .use(animateHandler("fadeOut", "fadeOut", fadeOutKeyframes))
  .use(animateHandler("fadeOutDown", "fadeOutDown", fadeOutDownKeyframes))
  .use(animateHandler("fadeOutDownBig", "fadeOutDownBig", fadeOutDownBigKeyframes))
  .use(animateHandler("fadeOutLeft", "fadeOutLeft", fadeOutLeftKeyframes))
  .use(animateHandler("fadeOutLeftBig", "fadeOutLeftBig", fadeOutLeftBigKeyframes))
  .use(animateHandler("fadeOutRight", "fadeOutRight", fadeOutRightKeyframes))
  .use(animateHandler("fadeOutRightBig", "fadeOutRightBig", fadeOutRightBigKeyframes))
  .use(animateHandler("fadeOutUp", "fadeOutUp", fadeOutUpKeyframes))
  .use(animateHandler("fadeOutUpBig", "fadeOutUpBig", fadeOutUpBigKeyframes))
  .use(animateHandler("backInUp", "backInUp", backInUpKeyframes))
  .use(animateHandler("backInDown", "backInDown", backInDownKeyframes))
  .use(animateHandler("backInLeft", "backInLeft", backInLeftKeyframes))
  .use(animateHandler("backInRight", "backInRight", backInRightKeyframes))
  .use(animateHandler("backOutUp", "backOutUp", backOutUpKeyframes))
  .use(animateHandler("backOutDown", "backOutDown", backOutDownKeyframes))
  .use(animateHandler("backOutLeft", "backOutLeft", backOutLeftKeyframes))
  .use(animateHandler("backOutRight", "backOutRight", backOutRightKeyframes))
  .init();
