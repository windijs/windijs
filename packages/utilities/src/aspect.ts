import { aspectRatioConfig, spacingConfig } from "@windijs/config";
import { configHandler, createUtility, cssHandler, genericHandler, numberHandler } from "@windijs/core";
import { css } from "@windijs/helpers";

import type { CSSObject, StyleObject } from "@windijs/helpers";

const aspectBase: CSSObject = {
  position: "relative",
  "> *": {
    position: "absolute",
    height: "100%",
    width: "100%",
    top: "0",
    right: "0",
    bottom: "0",
    left: "0",
  },
};

export default createUtility("aspect")
  .use(configHandler(aspectRatioConfig, "aspectRatio"))
  .case(
    "none",
    cssHandler({
      position: "static",
      paddingBottom: "0",
      "> *": {
        position: "static",
        height: "auto",
        width: "auto",
        top: "auto",
        right: "auto",
        bottom: "auto",
        left: "auto",
      },
    })
  )
  .case(
    "w",
    numberHandler<Record<keyof typeof spacingConfig, StyleObject> & Record<number, StyleObject>>(v =>
      css({
        "--w-aspect-w": v as string,
        paddingBottom: "calc(var(--w-aspect-h) / var(--w-aspect-w) * 100%)",
        ...aspectBase,
      })
    )
  )
  .case(
    "h",
    numberHandler<Record<keyof typeof spacingConfig, StyleObject> & Record<number, StyleObject>>(v =>
      css({
        "--w-aspect-h": v as string,
      })
    )
  )
  .use(
    genericHandler(v =>
      css({
        paddingBottom: +v * 100 + "%",
        ...aspectBase,
      })
    )
  )
  .init();
