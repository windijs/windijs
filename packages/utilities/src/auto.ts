import { gridAutoColumnsConfig, gridAutoRowsConfig } from "@windijs/config";
import { configHandler, createUtility } from "@windijs/core";

export default createUtility("auto")
  .case("rows", configHandler(gridAutoRowsConfig, "gridAutoRows"))
  .case("cols", configHandler(gridAutoColumnsConfig, "gridAutoColumns"))
  .init();
