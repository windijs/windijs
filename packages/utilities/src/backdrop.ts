import {
  blurConfig,
  brightnessConfig,
  contrastConfig,
  grayscaleConfig,
  hueRotateConfig,
  invertConfig,
  opacityConfig,
  saturateConfig,
  sepiaConfig,
} from "@windijs/config";
import { buildBackdropFilter, configHandler, createUtility, cssHandler } from "@windijs/core";
import { css } from "@windijs/helpers";

export default createUtility("backdrop")
  .use(
    cssHandler({
      "--w-backdrop-blur": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-brightness": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-contrast": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-grayscale": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-hue-rotate": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-invert": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-opacity": "var(--tw-empty,/*!*/ /*!*/)",
      "--w-backdrop-saturate": "var(--w-empty,/*!*/ /*!*/)",
      "--w-backdrop-sepia": "var(--w-empty,/*!*/ /*!*/)",
      "-webkit-backdrop-filter":
        "var(--w-backdrop-blur) var(--w-backdrop-brightness) var(--w-backdrop-contrast) var(--w-backdrop-grayscale) var(--w-backdrop-hue-rotate) var(--w-backdrop-invert) var(--w-backdrop-opacity) var(--w-backdrop-saturate) var(--w-backdrop-sepia)",
      backdropFilter:
        "var(--w-backdrop-blur) var(--w-backdrop-brightness) var(--w-backdrop-contrast) var(--w-backdrop-grayscale) var(--w-backdrop-hue-rotate) var(--w-backdrop-invert) var(--w-backdrop-opacity) var(--w-backdrop-saturate) var(--w-backdrop-sepia)",
    })
  )
  .case(
    "blur",
    configHandler(blurConfig, v => css({ "--w-backdrop-blur": `blur(${v})` }))
  )
  .case(
    "brightness",
    configHandler(brightnessConfig, v => css({ "--w-backdrop-brightness": `brightness(${v})` }))
  )
  .case(
    "contrast",
    configHandler(contrastConfig, v => css({ "--w-backdrop-contrast": `contrast(${v})` }))
  )
  .case(
    "grayscale",
    configHandler(grayscaleConfig, v => css({ "--w-backdrop-grayscale": `grayscale(${v})` }))
  )
  .case(
    "hueRotate",
    configHandler(hueRotateConfig, v => css({ "--w-backdrop-hue-rotate": `hue-rotate(${v})` }))
  )
  .case(
    "invert",
    configHandler(invertConfig, v => css({ "--w-backdrop-invert": `invert(${v})` }))
  )
  .case(
    "opacity",
    configHandler(opacityConfig, v => css({ "--w-backdrop-opacity": `opacity(${v})` }))
  )
  .case(
    "saturate",
    configHandler(saturateConfig, v => css({ "--w-backdrop-saturate": `saturate(${v})` }))
  )
  .case(
    "sepia",
    configHandler(sepiaConfig, v => css({ "--w-backdrop-sepia": `sepia(${v})` }))
  )
  .init(buildBackdropFilter);
