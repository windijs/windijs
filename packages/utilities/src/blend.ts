import { configHandler, createUtility } from "@windijs/core";

import { blendModeConfig } from "@windijs/config";

export default createUtility("blend")
  .use(configHandler(blendModeConfig, "mixBlendMode"))
  .init();
