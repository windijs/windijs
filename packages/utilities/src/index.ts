import animate from "./animate";
import aspect from "./aspect";
import auto from "./auto";
import backdrop from "./backdrop";
import bg from "./bg";
import blend from "./blend";
import blur from "./blur";
import border from "./border";
import brightness from "./brightness";
import col from "./col";
import colors from "./colors";
import container from "./container";
import contrast from "./contrast";
import decoration from "./decoration";
import delay from "./delay";
import divide from "./divide";
import dropShadow from "./dropShadow";
import duration from "./duration";
import ease from "./ease";
import fill from "./fill";
import filter from "./filter";
import flex from "./flex";
import font from "./font";
import from from "./from";
import gap from "./gap";
import grayscale from "./grayscale";
import grid from "./grid";
import h from "./h";
import hueRotate from "./hueRotate";
import hyphens from "./hyphens";
import image from "./image";
import indent from "./indent";
import invert from "./invert";
import leading from "./leading";
import list from "./list";
import m from "./m";
import mb from "./mb";
import ml from "./ml";
import mr from "./mr";
import mt from "./mt";
import mx from "./mx";
import my from "./my";
import opacity from "./opacity";
import origin from "./origin";
import outline from "./outline";
import overflow from "./overflow";
import overscroll from "./overscroll";
import p from "./p";
import pb from "./pb";
import perspect from "./perspect";
import pl from "./pl";
import placeholder from "./placeholder";
import pr from "./pr";
import preserve from "./preserve";
import pt from "./pt";
import px from "./px";
import py from "./py";
import ring from "./ring";
import rotate from "./rotate";
import rounded from "./rounded";
import row from "./row";
import saturate from "./saturate";
import scale from "./scale";
import sepia from "./sepia";
import shadow from "./shadow";
import skew from "./skew";
import space from "./space";
import sr from "./sr";
import stroke from "./stroke";
import tab from "./tab";
import text from "./text";
import to from "./to";
import tracking from "./tracking";
import transform from "./transform";
import transition from "./transition";
import translate from "./translate";
import via from "./via";
import w from "./w";
import write from "./write";

export {
  animate,
  aspect,
  auto,
  backdrop,
  bg,
  blend,
  blur,
  border,
  brightness,
  col,
  colors,
  container,
  contrast,
  decoration,
  delay,
  divide,
  dropShadow,
  duration,
  ease,
  fill,
  filter,
  flex,
  font,
  from,
  gap,
  grayscale,
  grid,
  h,
  hueRotate,
  hyphens,
  image,
  indent,
  invert,
  leading,
  list,
  m,
  mb,
  ml,
  mr,
  mt,
  mx,
  my,
  opacity,
  origin,
  outline,
  overflow,
  overscroll,
  p,
  pb,
  perspect,
  pl,
  placeholder,
  pr,
  preserve,
  pt,
  px,
  py,
  ring,
  rotate,
  rounded,
  row,
  saturate,
  scale,
  sepia,
  shadow,
  skew,
  space,
  sr,
  stroke,
  tab,
  text,
  to,
  tracking,
  transform,
  transition,
  translate,
  via,
  w,
  write,
};
