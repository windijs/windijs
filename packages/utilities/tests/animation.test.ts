import { animate } from "../src/animation";
import { unify } from "@windijs/helpers";

test("Animation", () => {
  expect(animate.none.css).toMatchSnapshot();
  expect(animate.bounce.css).toMatchSnapshot();
  expect(animate.flip.css).toMatchSnapshot();
  expect(animate.backInUp.css).toMatchSnapshot();
});

test("Build Animation Keyframes", () => {
  expect(unify(".animate-none", animate.none)).toMatchSnapshot();
  expect(unify(".animate-spin", animate.spin)).toMatchSnapshot();
  expect(unify(".animate-flip", animate.flip)).toMatchSnapshot();
});
