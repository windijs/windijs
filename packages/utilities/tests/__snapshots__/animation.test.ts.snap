// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation 1`] = `
Map {
  "-webkit-animation" => "none",
  "animation" => "none",
}
`;

exports[`Animation 2`] = `
Map {
  "@-webkit-keyframes bounce" => Object {
    "0%, 100%": Object {
      "-webkit-animation-timing-function": "cubic-bezier(0.8,0,1,1)",
      "-webkit-transform": "translateY(-25%)",
      "animationTimingFunction": "cubic-bezier(0.8,0,1,1)",
      "transform": "translateY(-25%)",
    },
    "50%": Object {
      "-webkit-animation-timing-function": "cubic-bezier(0,0,0.2,1)",
      "-webkit-transform": "translateY(0)",
      "animationTimingFunction": "cubic-bezier(0,0,0.2,1)",
      "transform": "translateY(0)",
    },
  },
  "@keyframes bounce" => Object {
    "0%, 100%": Object {
      "-webkit-animation-timing-function": "cubic-bezier(0.8,0,1,1)",
      "-webkit-transform": "translateY(-25%)",
      "animationTimingFunction": "cubic-bezier(0.8,0,1,1)",
      "transform": "translateY(-25%)",
    },
    "50%": Object {
      "-webkit-animation-timing-function": "cubic-bezier(0,0,0.2,1)",
      "-webkit-transform": "translateY(0)",
      "animationTimingFunction": "cubic-bezier(0,0,0.2,1)",
      "transform": "translateY(0)",
    },
  },
  "-webkit-animation" => "bounce 1s infinite",
  "animation" => "bounce 1s infinite",
}
`;

exports[`Animation 3`] = `
Map {
  "@-webkit-keyframes flip" => Object {
    "40%": Object {
      "-webkit-animation-timing-function": "ease-out",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg)",
      "animationTimingFunction": "ease-out",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg)",
    },
    "50%": Object {
      "-webkit-animation-timing-function": "ease-in",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg)",
      "animationTimingFunction": "ease-in",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg)",
    },
    "80%": Object {
      "-webkit-animation-timing-function": "ease-in",
      "-webkit-transform": "perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
      "animationTimingFunction": "ease-in",
      "transform": "perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
    },
    "from": Object {
      "-webkit-animation-timing-function": "ease-out",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg)",
      "animationTimingFunction": "ease-out",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg)",
    },
    "to": Object {
      "-webkit-animation-timing-function": "ease-in",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
      "animationTimingFunction": "ease-in",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
    },
  },
  "@keyframes flip" => Object {
    "40%": Object {
      "-webkit-animation-timing-function": "ease-out",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg)",
      "animationTimingFunction": "ease-out",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg)",
    },
    "50%": Object {
      "-webkit-animation-timing-function": "ease-in",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg)",
      "animationTimingFunction": "ease-in",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg)",
    },
    "80%": Object {
      "-webkit-animation-timing-function": "ease-in",
      "-webkit-transform": "perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
      "animationTimingFunction": "ease-in",
      "transform": "perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
    },
    "from": Object {
      "-webkit-animation-timing-function": "ease-out",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg)",
      "animationTimingFunction": "ease-out",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg)",
    },
    "to": Object {
      "-webkit-animation-timing-function": "ease-in",
      "-webkit-transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
      "animationTimingFunction": "ease-in",
      "transform": "perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg)",
    },
  },
  "-webkit-animation" => "flip",
  "animation" => "flip",
  "backfaceVisibility" => "visible",
}
`;

exports[`Animation 4`] = `
Map {
  "@-webkit-keyframes backInUp" => Object {
    "0%": Object {
      "-webkit-transform": "translateY(1200px) scale(0.7)",
      "opacity": "0.7",
      "transform": "translateY(1200px) scale(0.7)",
    },
    "100%": Object {
      "-webkit-transform": "scale(1)",
      "opacity": "1",
      "transform": "scale(1)",
    },
    "80%": Object {
      "-webkit-transform": "translateY(0px) scale(0.7)",
      "opacity": "0.7",
      "transform": "translateY(0px) scale(0.7)",
    },
  },
  "@keyframes backInUp" => Object {
    "0%": Object {
      "-webkit-transform": "translateY(1200px) scale(0.7)",
      "opacity": "0.7",
      "transform": "translateY(1200px) scale(0.7)",
    },
    "100%": Object {
      "-webkit-transform": "scale(1)",
      "opacity": "1",
      "transform": "scale(1)",
    },
    "80%": Object {
      "-webkit-transform": "translateY(0px) scale(0.7)",
      "opacity": "0.7",
      "transform": "translateY(0px) scale(0.7)",
    },
  },
  "-webkit-animation" => "backInUp",
  "animation" => "backInUp",
}
`;

exports[`Build Animation Keyframes 1`] = `
".animate-none {
  -webkit-animation: none;
  animation: none;
}"
`;

exports[`Build Animation Keyframes 2`] = `
"@-webkit-keyframes spin {
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  to {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes spin {
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  to {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.animate-spin {
  -webkit-animation: spin 1s linear infinite;
  animation: spin 1s linear infinite;
}"
`;

exports[`Build Animation Keyframes 3`] = `
"@-webkit-keyframes flip {
  from {
    -webkit-animation-timing-function: ease-out;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg);
    animation-timing-function: ease-out;
  }
  40% {
    -webkit-animation-timing-function: ease-out;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg);
    animation-timing-function: ease-out;
  }
  50% {
    -webkit-animation-timing-function: ease-in;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg);
    animation-timing-function: ease-in;
  }
  80% {
    -webkit-animation-timing-function: ease-in;
    -webkit-transform: perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    transform: perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    animation-timing-function: ease-in;
  }
  to {
    -webkit-animation-timing-function: ease-in;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    animation-timing-function: ease-in;
  }
}

@keyframes flip {
  from {
    -webkit-animation-timing-function: ease-out;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, -360deg);
    animation-timing-function: ease-out;
  }
  40% {
    -webkit-animation-timing-function: ease-out;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -190deg);
    animation-timing-function: ease-out;
  }
  50% {
    -webkit-animation-timing-function: ease-in;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 150px) rotate3d(0, 1, 0, -170deg);
    animation-timing-function: ease-in;
  }
  80% {
    -webkit-animation-timing-function: ease-in;
    -webkit-transform: perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    transform: perspective(400px) scale3d(0.95, 0.95, 0.95) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    animation-timing-function: ease-in;
  }
  to {
    -webkit-animation-timing-function: ease-in;
    -webkit-transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    transform: perspective(400px) scale3d(1, 1, 1) translate3d(0, 0, 0) rotate3d(0, 1, 0, 0deg);
    animation-timing-function: ease-in;
  }
}

.animate-flip {
  -webkit-animation: flip;
  animation: flip;
  backface-visibility: visible;
}"
`;
